templates/sig/prism/node.rbs.erb

```diff
@@ -0,0 +1,70 @@
+module Prism
+  class Node
+    @newline: bool
+
+    attr_reader location: Location
+
+    def child_nodes: () -> Array[node?]
+    def newline?: () -> bool
+    def set_newline_flag: (Array[bool] newline_marked) -> void
+    def slice: () -> String
+    def pretty_print: (untyped q) -> untyped
+    def to_dot: () -> String
+  end
+
+  interface _NodeImpl
+    def accept: (_Visitor) -> void
+    def comment_targets: () -> Array[node | Location]
+    def compact_child_nodes: () -> Array[node]
+    def inspect: (?NodeInspector) -> String
+  end
+
+  type node = Node & _NodeImpl
```

Comment:
This class and interface structure ended up being necessary because methods taking a Node actually only can take a concrete subclass of Node, since they might call accept or inspect, etc.

If you're okay with making Node a module instead of a class, then I think we can simplify this since no instances of a Node will exist at runtime.

Another option if you want to keep Node as a class: If we split up the sigs into public (included in the gem) vs private, I'll move `class Node` definition into the private side. Then make a `_Node` interface that has all the methods currently in `class Node` and `interface _NodeImpl` and only the subclasses will `include _Node`. Then we can get rid of this type alias and use `_Node` everywhere.

Reply:
I think I would very much like a public/private split like you're saying here.